src/borg/fuse.py

```diff
@@ -259,10 +260,17 @@ def _create_filesystem(self):
                     self.pending_archives[archive_inode] = archive.name
 
     def get_item(self, inode):
+        item = self._read_cache.get(inode)
+        if item:
+            return item        
         try:
-            return self._items[inode]
+            item = self._items[inode]
+            self._read_cache[inode] = item
+            return item
         except KeyError:
-            return self.cache.get(inode)
+            item = self.cache.get(inode)
+            self._read_cache[inode] = item 
```

Comment:
otoh, this makes sense as there is some overhead in self.cache.get() - although it internally also uses a cache for chunks.